zebra-consensus/src/chain.rs

```diff
@@ -50,11 +50,15 @@ where
     type Future =
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
-    fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        // We don't expect the verifiers to exert backpressure on our
-        // users, so we don't need to call the verifier's `poll_ready` here.
-        // (And we don't know which verifier to choose at this point, anyway.)
-        Poll::Ready(Ok(()))
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        match (self.checkpoint.poll_ready(cx), self.block.poll_ready(cx)) {
+            // Unready if either service is unready.
+            (Poll::Pending, _) | (_, Poll::Pending) => Poll::Pending,
```

Comment:
This will swallow errors if one of the services returns pending and the other returns error. We either need to more exhaustively match things or reorder the arms so the error match happens first.

Reply:
Fixed in 2e41de5b, I'll squash the fixup commit as soon as it looks OK to you.

Reply:
looks good, squash away.